VERSION-RANGE-SPEC.rst

```diff
@@ -0,0 +1,658 @@
+  ``vers`` simplified notation defined here.
+
+- by convention the versioning scheme should be the same string as the Package
+  URL package type for a given package ecosystem. It is OK to have other schemes
+  beyond the purl type and even schemes that are specific to a single package.
+
+The ``<versioning-scheme>`` is followed by a slash "/".
+
+
+``<version-constraint>``
+----------------------------
+
+After the ``<versioning-scheme>`` and "/" there are one or more
+``<version-constraint>`` separated by a pipe "|". The pipe "|" means that
+**any** of these constraints must be satisfied for a version to be resolved as
+within this version range.
+
+Each  ``<version-constraint>`` of this pipe-separated list can be either a
+single constraint or a list of constraints separated in turn by an comma "," as
+in ``1.2.3|>=2.0.0,<5.0.0``.
+
+Multiple ``<version-constraint>`` combined with a comma means that **all** these
+constraints must be satisfied for a version to be resolved as contained in this
+range.
+
+Each simple version constraint has this syntax::
+
+    <comparator><version>
+
+The ``<comparator>`` is one of these comparison operators:
+
+- "=": Version equality comparator. It is the default and implied if not
+  present and means that a version must be equal to the provided version.
+  For example: "=1.2.3". It must be omitted in the canonical representation.
+  Equality is based on the equality of two version strings and is typically
+  not versioning scheme-specific, though some schemes such as pypi PEP440
+  may require a version string normalization before testing for equality.
+
+- "!=": Version exclusion or inequality comparator. This means a version must
+  not be equal to the provided version and this version must be excluded from
+  the range. For example: "!=1.2.3" means that version "1.2.3" is excluded.
+
+- "<", "<=": Less than or less-or-equal version comparators points to all
+  versions less than or equal to the provided version. For example "<=1.2.3"
+  means less than or equal to "1.2.3". 
+
+- ">", ">=": Greater than or greater-or-equal version comparators points to
+  all versions greater than or equal to the provided version. For example
+  ">=1.2.3" means greater than or equal to "1.2.3".
+
+- The way two version strings are compared using these comparators is defined
+  by the ``<versioning-scheme>``.
+
+- The structure and meaning of a version string such as "1.2.3" is defined by
+  the ``<versioning-scheme>``. For instance, ``semver`` defines three
+  dot-separated segments name major, minor and patch.
+
+- The special star "*" ``<version-constraint>`` matches any version. This star
+  constraint must be used **alone** in a version range, exclusive of any other
+  constraint. For example "vers:deb/\*" resolves to any version of a Debian
+  package.
+
+- The way each of these comparators work when doing a version comparison is
+  specific to a versioning scheme.
+
+
+Examples
+~~~~~~~~~
+
+TODO.
```

Comment:
It _might_ make sense to have examples in another file so that implementations of the spec can utilize the examples in their unit testing.

Reply:
Yes! I added narrated examples for now. I will add a JSON file with language-independent tests to guide implementers. :+1: